benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -29,103 +29,27 @@ def stressng_pod(self, name: str = ''):
-            else:
-                self.__es_index = 'stressng-results'
-            environment_variables.environment_variables_dict['kind'] = self.__kind
-            self._oc.create_pod_sync(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_pod.__name__}.yaml'), pod_name=f'{self.__workload}-workload')
-            self._oc.wait_for_initialized(label='app=stressng_workload', workload=self.__workload)
-            self._oc.wait_for_ready(label='app=stressng_workload', workload=self.__workload)
-            self.__status = self._oc.wait_for_pod_completed(label='app=stressng_workload', workload=self.__workload)
-            self.__status = 'complete' if self.__status else 'failed'
-            # system metrics
-            if environment_variables.environment_variables_dict['system_metrics'] == 'True':
-                self.system_metrics_collector(workload=self.__workload)
-            # save run artifacts logs
-            run_artifacts_url = self._create_run_artifacts(workload=self.__workload)
-            if self._es_host:
-                # verify that data upload to elastic search according to unique uuid
-                ids = self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._oc.get_long_uuid(workload=self.__workload))
-                # update metadata
-                for id in ids:
-                    self._update_elasticsearch_index(index=self.__es_index, id=id, kind=self.__kind, status=self.__status, run_artifacts_url=run_artifacts_url)
-            self._oc.delete_pod_sync(
-                yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_pod.__name__}.yaml'),
-                pod_name=f'{self.__workload}-workload')
-        except ElasticSearchDataNotUploaded as err:
-            self.tear_down_pod_after_error(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_pod.__name__}.yaml'),
-                                           pod_name=f'{self.__workload}-workload')
-            raise err
-        except Exception as err:
-            # save run artifacts logs
-            run_artifacts_url = self._create_run_artifacts(workload=self.__workload)
-            self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status='failed', run_artifacts_url=run_artifacts_url)
-            self.tear_down_pod_after_error(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.stressng_pod.__name__}.yaml'),
-                                           pod_name=f'{self.__workload}-workload')
-            raise err
+        if name == '':
+            name = self.stressng_pod.__name__
+        run = StressngPod()
+        run.stressng_pod(name=name)
 
-    # The Kata workloads should not be decorated with
     @logger_time_stamp
     def stressng_kata(self):
         """
         This method run stressng kata workload
         :return:
         """
-        self.stressng_pod(self.stressng_kata.__name__)
+        self.stressng_pod(name=self.stressng_kata.__name__)
```

Comment:
This file should not contain workload-specific code.

Reply:
@RobertKrawitz thx for this great feedback , remove all workloads from this module and create a dynamic calls.